sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/data/FollowingViewportStateOptions.kt

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.maps.plugin.viewport.experimental.data
+
+import com.mapbox.maps.plugin.viewport.DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS
+import java.util.Objects
+
+/**
+ * Options that impact the [FollowingViewportState].
+ */
+class FollowingViewportStateOptions private constructor(
+  /**
+   * The default zoom that will be generated for camera following frames.
+   *
+   * Defaults to `16.35`.
+   */
+  val defaultZoom: Double,
+  /**
+   * The default pitch that will be generated for following camera frames.
+   *
+   * Defaults to `45.0` degrees.
+   */
+  val defaultPitch: Double,
+  /**
+   * The maximum duration between frames in ms.
+   *
+   * Defaults to [DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS] ms
+   */
+  val frameTransitionMaxDurationMs: Long
+) {
+  /**
+   * Returns a builder that created the [FollowingViewportStateOptions]
+   */
+  fun toBuilder() = Builder().defaultZoom(defaultZoom).defaultPitch(defaultPitch)
+    .frameTransitionMaxDurationMs(frameTransitionMaxDurationMs)
+
+  /**
+   * Indicates whether some other object is "equal to" this one.
+   */
+  override fun equals(other: Any?) = other is FollowingViewportStateOptions &&
+    defaultZoom == other.defaultZoom &&
+    defaultPitch == other.defaultPitch &&
+    frameTransitionMaxDurationMs == other.frameTransitionMaxDurationMs
+
+  /**
+   * Returns a hash code value for the object.
+   */
+  override fun hashCode() = Objects.hash(defaultZoom, defaultPitch, frameTransitionMaxDurationMs)
+
+  /**
+   * Returns a String for the object.
+   */
+  override fun toString() =
+    "FollowingViewportStateOptions(defaultZoom=$defaultZoom, defaultPitch=$defaultPitch, frameTransitionMaxDurationMs=$frameTransitionMaxDurationMs)"
+
+  /**
+   * Builder for [FollowingViewportStateOptions]
+   */
+  class Builder {
+    private var defaultZoom: Double = 16.35
+    private var defaultPitch: Double = 45.0
+    private var frameTransitionMaxDuration: Long = DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS
+
+    /**
+     * The default zoom that will be generated for camera following frames.
+     *
+     * Defaults to `16.35`.
+     */
+    fun defaultZoom(zoom: Double) = apply {
+      this.defaultZoom = zoom
+    }
+
+    /**
+     * The default pitch that will be generated for following camera frames.
+     *
+     * Defaults to `45.0` degrees.
```

Comment:
What about using named constants similarly to `DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS` below?

Reply:
Addressed in https://github.com/mapbox/mapbox-maps-android/pull/1034